internal/controllers/scheduling/op.go

```diff
@@ -19,20 +19,34 @@ type op struct {
 	Synthesizer *apiv1.Synthesizer
 	Composition *apiv1.Composition
 	Reason      opReason
+	NotBefore   time.Time
 
 	Dispatched time.Time
 
 	id               uuid.UUID // set when patch is built
 	synthRolloutHash []byte    // memoized
 }
 
-func newOp(synth *apiv1.Synthesizer, comp *apiv1.Composition) *op {
+func newOp(synth *apiv1.Synthesizer, comp *apiv1.Composition, nextCooldownSlot time.Time) *op {
 	o := &op{Synthesizer: synth, Composition: comp}
 
 	var ok bool
 	o.Reason, ok = classifyOp(synth, comp)
 	if !ok {
-		return nil
+		syn := comp.Status.InFlightSynthesis
+		if syn == nil || syn.Canceled == nil {
+			return nil
+		}
+
+		// Retry canceled syntheses with exponential backoff
+		backoff := time.Duration(float64(syn.Canceled.Sub(syn.Initialized.Time)) * float64(syn.Attempts) * 1.5)
```

Comment:
Why not use the pod timeout instead of cancelled - initialized?

Reply:
This is completely pedantic but I was thinking about what happens if the synth's pod timeout changes. I'd expect existing syntheses to back off using the timeout specified by their `observedSynthesizerGeneration` - not the new value.

Reply:
That's a valid reason, I was just curious about why one over the other. A (welcome?) side effect of this is that it might add a bit of jitter.